### PR TITLE
DEV-22511: add licenses api group with statistics sub-section

### PR DESCRIPTION
**Pull request by george-haddad**
Add a new section for the `/licenses` end-point with a subsection on statistics.

### CODE REVIEW

**Review by tkrisztian95 — COMMENTED** [review]
Looks good. Nice and informative. Especially the table with the statistics field descriptions. It was a good idea to add as a summary.

File: apiary.apib [hunk 1]
```diff
@@ -3185,6 +3185,85 @@ noted in the Create User API section.
...
+- All `concurrent` licensed users will be `existing` if a license exists
+
+Details of the statistics attributes in the table below:
+
+| License Type | Sum Attribute | Description                   |
+|:-------------|---------------|-------------------------------|
+| `named`      | licensed      | Total number of licenses      |
+| `named`      | active        | Total active licenses         |
+| `named`      | inactive      | Total inactive licenses       |
+| `named`      | available     | Total available licenses (`licensed` - `active`) |
```
[the remark this answers is not in the document]
Comment (george-haddad) [reply apiary.apib:3214]: cool with me.

File: apiary.apib [hunk 2]
```diff
@@ -3185,6 +3185,90 @@ noted in the Create User API section.
...
+                "title": "Invalid authentication",
+                "status": 401
+            }
+        }
+
++ Response 403 (application/json)
+
+        // when there is insufficient privileges to fetch license statistics
+        {
+        "links": {},
```
Comment (tkrisztian95) [inline apiary.apib:3262]: Hmm, I think this should be shifted a bit more to the right. Could you also fix this indentation? (I think there are a bunch of places due to the copy&paste)
```suggestion
            "links": {},
```

File: apiary.apib [hunk 3]
```diff
@@ -3185,6 +3185,85 @@ noted in the Create User API section.
...
+**Note:**
+- Must have `UserAndRoles.editUser` privileges to perform this request
+
++ Response 200 (application/json)
+
+    + Attributes (LicenseStatistics)
+
+    + Body
+
+            {
```
[the remark this answers is not in the document]
Comment (george-haddad) [reply apiary.apib:3229]: It does now with the new changes

File: apiary.apib (hunk 2 above)
Comment (george-haddad) [reply apiary.apib:3262]: yeah, for some reason the copy/paste is not preserving the indentations.

File: apiary.apib [hunk 4]
```diff
@@ -3185,6 +3185,85 @@ noted in the Create User API section.
...
+# Group Licenses API
+
+License information is provided by the following set of APIs.
+
+## Statistics [/api/v1/licenses/statistics]
+
+### Get License Statistics for Users [GET]
+
+Returns license statistics for the running instance showing the sum of license types `named` and `concurrent`.
+The Built-in users such as `Admin`, `TermBrowser`, `TermTargetAccess`, `TermContribution`, `TermApiAccess` and `AnalyticsReadOnlyUser`
```
[the remark this answers is not in the document]
Comment (george-haddad) [reply apiary.apib:3197]: yeah it can be moved.

File: apiary.apib [hunk 5]
```diff
@@ -3185,6 +3185,85 @@ noted in the Create User API section.
...
+are included in the statistics calculation along with all the non-builtin users, however the following conditions are checked
+that will ultimately decide the user to be included in the calculation:
+
+- Must be one of the following License Types `named` or `concurrent`
+- For `named` license they must be activated for `Checking` session type
```
[the remark this answers is not in the document]
Comment (george-haddad) [reply apiary.apib]: done.

File: apiary.apib [hunk 6]
```diff
@@ -3185,6 +3185,85 @@ noted in the Create User API section.
...
+|:-------------|---------------|-------------------------------|
+| `named`      | licensed      | Total number of licenses      |
+| `named`      | active        | Total active licenses         |
+| `named`      | inactive      | Total inactive licenses       |
+| `named`      | available     | Total available licenses (`licensed` - `active`) |
+| `concurrent` | licensed      | Total number of licenses      |
+| `concurrent` | existing      | Total number of licenses used |
+
+**Note:**
+- Must have `UserAndRoles.editUser` privileges to perform this request
```
[the remark this answers is not in the document]
Comment (george-haddad) [reply apiary.apib:3221]: I am not sure what to say about it aside from what I know that concurrent licensing is no longer being sold, but I guess whoever uses this understands this.

[the remark this answers is not in the document]
Comment (tkrisztian95) [reply apiary.apib:3221]: How about something like this:

> You can only see `concurrent` license use numbers if you’ve installed a license that is configured with both named and concurrent user limits.

[the remark this answers is not in the document]
Comment (george-haddad) [reply apiary.apib:3221]: No I think like that it is relevant and with the excerpt you found in the official docs it makes sense to put that.

**Review by tkrisztian95 — APPROVED** [review]
Looks good!

**Pull request merged**